eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala

```diff
@@ -24,13 +28,33 @@ case class GetInfoResult(
 case class PeerInfo(
     nodeId: NodeId,
     state: PeerState,
-    //address: String,
+    address: Option[String],
     channels: Int)
 
+sealed trait ChannelInfoT {
```

Comment:
I'm not sure what to call this...
The issue here is that the result of querying for channel info varies based on what state the channel is in. Here I've only implemented the `ChannelState.NORMAL` and the default cases, haven't looked into the details of the other cases.

Reply:
It seems to me the abstract type should be `ChannelInfo` and then the thing you have named `ChannelInfo` now should be renamed to `NormalChannelInfo`? As in the channel is in a normal state? Maybe i am misunderstaning what is going on here. You could also do 

```
sealed abstract class ChannelInfo {
  def nodeId: NodeId
  def channelId: FundedChannelId
  def localMsat: MilliSatoshis
  def remoteMsat: MilliSatoshis
  def state: ChannelState
}

case class BaseChannelInfo(
    nodeId: NodeId,
    channelId: FundedChannelId,
    localMsat: MilliSatoshis,
    remoteMsat: MilliSatoshis,
    state: ChannelState
) extends ChannelInfo

case class OpenChannelInfo(
    nodeId: NodeId,
    shortChannelId: ShortChannelId,
    channelId: FundedChannelId,
    localMsat: MilliSatoshis,
    remoteMsat: MilliSatoshis
) extends ChannelInfo {
  override val state: ChannelState.NORMAL.type = ChannelState.NORMAL
}
```

I'm not sure if I understand what the point of `BaseChannelInfo` is though.

Reply:
Side note: let's make sure we use abstract classes where possible for older java compatability. If we wan't use scala 2.12 (which android doesn't support) I dont think we get a very strong guarantee that a trait is going to work on android

Reply:
The point of `BaseChannelInfo` is to catch all responses where `state != "NORMAL"`